src/config/Config.py

```diff
@@ -0,0 +1,81 @@
+import sys
+import json
+import os
+from datetime import datetime
+from dateutil.parser import parse
+
+from enum import Enum
+
+class BoundsResult(Enum):
+    VALID = 1     #value is within range
+    MAX_ERROR = 2 #value is greater than MAX
+    MIN_ERROR = 3 #value is less than MIN
+
+class Config:
+    def __init__(self):
+        self._data = {}
+
+    def load(self, filename="./src/config/config.json", debug=False):
+        with open(filename) as f:
+            self._data = json.load(f)
+
+    def ingest_env(self):
+        if "PIPELINE_USER" in os.environ:
+            self._data["pipeline_user"] = os.environ["PIPELINE_USER"]
+        if "PIPELINE_PASSWD" in os.environ:
+            self._data["pipeline_passwd"] = os.environ["PIPELINE_PASSWD"]
+        if "PIPELINE_HOSTNAME" in os.environ:
+            self._data["pipeline_hostname"] = os.environ["PIPELINE_HOSTNAME"]
+        if "PIPELINE_DB_NAME" in os.environ:
+            self._data["pipeline_db_name"] = os.environ["PIPELINE_DB_NAME"]
+
+    def set_value(self, name, val):
+        self._data[name] = val
+
+    def get_value(self, val):
+        if val in self._data:
+            return self._data[val]
```

Comment:
It shouldn't ever keyerror. It's already checking to make sure val is in the _data dict. If val isn't in self._data it will return None implicitly. I could make it explicit with a return None if that's more clear but as far as I can tell the behavior will be the same.

Reply:
Oh I see. Btw, why is the section that calls ingest_env in main trying to catch keyerror? It doesn't seem like ingest_value or get_value would return keyerror.

Reply:
I think you are looking at an older version, I removed the unnecessary KeyError exception handling in a refactor after I better understood what Sawyer's code was doing in main.py

Reply:
Yep, my bad.